integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/receipt/DpoReceiptStatus.java

```diff
@@ -0,0 +1,42 @@
+package no.difi.meldingsutveksling.receipt;
+
+import no.difi.meldingsutveksling.kvittering.xsd.Kvittering;
+
+public enum DpoReceiptStatus implements ReceiptStatus {
+    AAPNING("Åpningskvittering"),
+    LEVERING("Leveringskvittering"),
+    MOTTAK("Mottakskvittering"),
+    VARSLINGFEILET("Varsling feilet"),
+    RETURPOST("Returpost"),
+    UKJENT("Ukjent");
+
+    private String status;
+
+    DpoReceiptStatus(String status) {
+        this.status = status;
+    }
+
+    @Override
+    public String getStatus() {
+        return null;
+    }
+
+    public static DpoReceiptStatus of(Kvittering kvittering) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "of" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)